ray-operator/controllers/ray/common/service.go

```diff
@@ -307,9 +307,10 @@ func BuildHeadlessServiceForRayCluster(rayCluster rayv1.RayCluster) (*corev1.Ser
 			Labels:    labels,
 		},
 		Spec: corev1.ServiceSpec{
-			ClusterIP: "None",
-			Selector:  selectorLabels,
-			Type:      corev1.ServiceTypeClusterIP,
+			ClusterIP:                "None",
+			Selector:                 selectorLabels,
+			Type:                     corev1.ServiceTypeClusterIP,
+			PublishNotReadyAddresses: true,
```

Comment:
Can you add a comment here for why we use `PublishNotReadyAddresses: true`? I think the Kubernetes docs has a good example:

```
 The primary use case for setting this field is for a StatefulSet's Headless Service to propagate SRV DNS records for its Pods for the purpose of peer discovery
```

We can say something similar that the headless service is used for peer discovery for TPU workers and should not be dependant on proxy actor placement

Reply:
Done in [b020382](https://github.com/ray-project/kuberay/pull/2375/commits/b020382ad4a41279e5d8d6add8db707b7684e5bd).